### PR TITLE
[WIP] Support multiple views

### PR DESCRIPTION
**Pull request by cbrewster**
Here is the initial approach behind allowing multiple views.

I am not happy with the ways I have to avoid making borrowck angry, so that will need to be cleaned up. I think this should be a good general direction though.

This is also waiting on cursor OT.

### CODE REVIEW

File: rust/core-lib/src/tabs.rs [hunk 1]
```diff
@@ -239,18 +263,18 @@ impl<W: Write + Send + 'static> Tabs<W> {
 }
 
 impl<W: Write> TabCtx<W> {
-    pub fn update_view(&self, view_id: &str, update: &Value) {
+    pub fn update_view(&self, view_id: &ViewIdentifier, update: &Value) {
         self.rpc_peer.send_rpc_notification("update",
             &json!({
-                "view_id": view_id,
+                "view_id": view_id.clone().into_value(),
```
Comment (cmyr) [inline rust/core-lib/src/tabs.rs]: maybe we just implement serialize/deserialize for `ViewIdentifier`?

Comment (cbrewster) [inline rust/core-lib/src/tabs.rs]: Yeah, will do. Was just slopping it together and didnt want to look at docs for implementing Serialize/Deserialize manually

**Comment by cbrewster** [comment]
I think there is an argument to be made about storing the id vs passing the id to different places. I think the less state we store, the better off we will be in the future. I could see storing the view_id on Editor causing some hard to track down bugs.

**Comment by cmyr** [comment]
yea, I do appreciate that general philosophy. I don't have really strong feelings either way; the main thing that influences me is how we have, so far, done some of this state-stashing (around cursor pos, scrolling, and applying edits) so this sort of seems to fall in line with that. 🤷‍♂️

**Comment by cbrewster** [comment]
Not sure if I will have the time to continue working on this in the near future.

**Pull request closed without merging**